discovery-handlers/debug-echo/src/discovery_handler.rs

```diff
@@ -76,53 +90,56 @@ impl DiscoveryHandler for DiscoveryHandlerImpl {
                     break;
                 }
 
+                //Standardize the procedures as same as the method used in OPC protocol discovery handler
+                //1. get a filtered list of current dicovery
+                //2. Query all devices' external info (if there is query uri configuration) 
+                //3. compare current Devices list with the Devices list of last discovery iteration
+                //4. check if there is no change (two lists have same length, and every component in current list is in previous list). If there is discrepancy, return full list of Devices
+                
+                let mut latest_discovered_list:Vec<Device>= Vec::new();
                 let availability =
                     fs::read_to_string(DEBUG_ECHO_AVAILABILITY_CHECK_PATH).unwrap_or_default();
                 trace!(
                     "discover -- debugEcho devices are online? {}",
                     !availability.contains(OFFLINE)
                 );
-                if (availability.contains(OFFLINE) && !offline) || offline && first_loop {
-                    if first_loop {
-                        first_loop = false;
-                    }
-                    // If the device is now offline, return an empty list of instance info
-                    offline = true;
-                    if let Err(e) = discovered_devices_sender
-                        .send(Ok(DiscoverResponse {
-                            devices: Vec::new(),
-                        }))
-                        .await
-                    {
-                        error!("discover - for debugEcho failed to send discovery response with error {}", e);
-                        if let Some(sender) = register_sender {
-                            sender.send(()).await.unwrap();
+
+                if  !availability.contains(OFFLINE) {
+                    let device_query_requests = descriptions.iter()
+                    .map(|description| {
+                        let mut properties = HashMap::new();
+                        properties.insert(
+                            super::DEBUG_ECHO_DESCRIPTION_LABEL.to_string(),
+                            String::from(description),
+                        );
+                        let query_body = QueryDevicePostBody{
+                            id:String::from(description),
+                            protocol: "DebugEcho".to_string()
```

Comment:
maybe use DISCOVERY_HANDLER_NAME here?

Reply:
you are right that I shall use the constants. I will change.